pathtools/path.py

```diff
@@ -73,7 +73,7 @@ def walk(path, topdown=topdown, followlinks=followlinks):
             try:
                 yield next(os.walk(path, topdown=topdown, followlinks=followlinks))
             except NameError:
-                yield os.walk(path, topdown=topdown, followlinks=followlinks).next() #IGNORE:E1101
+                yield next(os.walk(path, topdown=topdown, followlinks=followlinks)) #IGNORE:E1101
```

Comment:
Note that this is now exactly the same line as inside the "try". I think this was here for Python 2.6 compatibility. Instead of this change, the except clause should simply be removed.